cmd/kops/create_cluster.go

```diff
@@ -734,17 +735,27 @@ func RunCreateCluster(ctx context.Context, f *util.Factory, out io.Writer, c *Cr
 		}
 
 		if autoloadSSHPublicKeys {
-			// Load from default location, if found
-			sshPublicKeyPath := "~/.ssh/id_rsa.pub"
-			c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
-			if err != nil {
+			// Load from default locations, if found
+			sshPublicKeyPaths := []string{
+				"~/.ssh/id_rsa.pub",
+				"~/.ssh/id_ed25519.pub",
+			}
+			var merr error
+			for _, sshPublicKeyPath := range sshPublicKeyPaths {
+				c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
+				if err == nil {
+					break
+				}
 				// Don't wrap file-not-found
 				if os.IsNotExist(err) {
 					klog.V(2).Infof("ssh key not found at %s", sshPublicKeyPath)
 				} else {
-					return fmt.Errorf("error reading SSH key file %q: %v", sshPublicKeyPath, err)
+					merr = multierr.Append(merr, err)
 				}
 			}
+			if merr != nil && len(c.SSHPublicKeys) == 0 {
+				return fmt.Errorf("error reading SSH public key files %q: %v", sshPublicKeyPaths, merr)
+			}
```

Comment:
The new code will incorrectly fall through if `os.IsNotExist(err)` on both files.